test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/AzureBlobSinkTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Internal;
+using Microsoft.WindowsAzure.Storage;
+using Moq;
+using Serilog;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.Formatting.Display;
+using Serilog.Parsing;
+using Xunit;
+
+namespace Microsoft.Extensions.Logging.AzureWebApps.Test
+{
+    public class AzureBlobSinkTests
+    {
+        [Fact]
+        public async Task WritesMessagesInBatches()
+        {
+            var blob = new Mock<ICloudAppendBlob>();
+            var buffers = new List<byte[]>();
+            blob.Setup(b => b.OpenWriteAsync()).Returns(() => Task.FromResult((Stream)new TestMemoryStream(buffers)));
+
+            var logger = CreateLogger(name => blob.Object);
+
+            for (int i = 0; i < 5; i++)
+            {
+                logger.Information("Text " + i);
+            }
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
You can probably poll the buffer until you get the expected number of messages. (with a timeout, of course)